### PR TITLE
[ISSUE #1870] Adopt OpenJDK 11

### PR DESCRIPTION
**Pull request by HashZhang**
Adopted OpenJDK 11

related to Issue #1870

add JVM flags for OpenJDK 11 and keep things compatible to JDK 8

### CODE REVIEW

**Comment by HashZhang** [comment]
或者要不开一个 OpenJDK 11 的分支，我有一些代码还想改

**Comment by zongtanghu** [comment]
@HashZhang 这样子吧，我给你单独开一个远端分支，专门整jdk11的？

**Comment by HashZhang** [comment]
> 我给你单独开一个远端分支，专门整jdk11的？

好哇，感谢~

**Comment by zongtanghu** [comment]
@HashZhang I have already checkout a new branch https://github.com/apache/rocketmq/tree/feature_support_java11 in apache rocketmq repo.Please resolve your code conflicts in your local enviroment.

**Comment by zongtanghu** [comment]
How can I keep in touch with you, dingding or wechat? @HashZhang

**Comment by HashZhang** [comment]
@zongtanghu Wechat Please, thanks

**Comment by zongtanghu** [comment]
> @zongtanghu Wechat Please, thanks

what's your wechat account?

**Comment by zongtanghu** [comment]
那你本地修改后，直接往我给你新创建的分支（https://github.com/apache/rocketmq/tree/feature_support_java11 ）上提pr吧，然后这个pr要不然我先close?你看呢？

**Comment by zongtanghu** [comment]
而且，你提pr以后注意下，一般来说都是往apache:rocketmq/develop分支上提的；

**Comment by HashZhang** [comment]
Wechat：2571656055
好的，知道啦，谢谢

File: distribution/bin/runbroker.sh [hunk 1]
```diff
@@ -64,18 +64,37 @@ choose_gc_log_directory()
...
-JAVA_OPT="${JAVA_OPT} -XX:+AlwaysPreTouch"
-JAVA_OPT="${JAVA_OPT} -XX:MaxDirectMemorySize=15g"
-JAVA_OPT="${JAVA_OPT} -XX:-UseLargePages -XX:-UseBiasedLocking"
-JAVA_OPT="${JAVA_OPT} -Djava.ext.dirs=${JAVA_HOME}/jre/lib/ext:${BASE_DIR}/lib"
-#JAVA_OPT="${JAVA_OPT} -Xdebug -Xrunjdwp:transport=dt_socket,address=9555,server=y,suspend=n"
-JAVA_OPT="${JAVA_OPT} ${JAVA_OPT_EXT}"
-JAVA_OPT="${JAVA_OPT} -cp ${CLASSPATH}"
+version=$($JAVA -version 2>&1 | awk -F '"' '/version/ {print $2}')
+version=${version%.*.*}
+if [[ "$version" -eq 11 ]]; then
```
Comment (RongtongJin) [inline distribution/bin/runbroker.sh:69]: IMO, it would be better to use ‘-ge’ instead of  ‘-eq’.

Comment (HashZhang) [reply distribution/bin/runbroker.sh:69]: Noted, thanks

Comment (HashZhang) [reply distribution/bin/runbroker.sh:69]: But there are some Flags deprecated in Java 12 and versions above, I need to check them

**Comment by zongtanghu** [comment]
I have already join you in wechat.

**Pull request closed without merging**